OpenRA.Mods.Common/Traits/World/Locomotor.cs

```diff
@@ -186,11 +189,20 @@ public int MovementSpeedForCell(CPos cell)
 			return terrainInfos[index].Speed;
 		}
 
-		public short MovementCostToEnterCell(Actor actor, CPos destNode, BlockedByActor check, Actor ignoreActor)
+		public short MovementCostToEnterCell(Actor actor, CPos destNode, CVec direction, BlockedByActor check, Actor ignoreActor)
 		{
 			if (!world.Map.Contains(destNode))
 				return short.MaxValue;
 
+			if (Info.CheckAdjacentOnDiagonal && direction.X != 0 && direction.Y != 0)
+			{
+				if (MovementCostToEnterCell(actor, destNode - new CVec(0, direction.Y), new CVec(direction.X, 0), check, ignoreActor) == short.MaxValue)
+					return short.MaxValue;
```

Comment:
```suggestion
					return PathGraph.MovementCostForUnreachableCell;
```

since https://github.com/OpenRA/OpenRA/pull/19993